metagraph/src/cli/query.cpp

```diff
@@ -166,96 +166,99 @@ Json::Value get_label_as_json(const std::string &label) {
-            sig_obj["score"] =
-                    Json::Value(anno_graph.score_kmer_presence_mask(kmer_presence_mask));
-
-            label_obj[SIGNATURE_FIELD] = sig_obj;
-            root["results"].append(label_obj);
+            sig_obj["score"] = Json::Value(anno_graph.score_kmer_presence_mask(kmer_presence_mask));
+            // Add kmer_counts calculated using bitmask
+            label_obj[KMER_COUNT_FIELD] = Json::Value(sdsl::util::cnt_one_bits(kmer_presence_mask));
         }
-    } else if (std::holds_alternative<LabelQuantileVec>(result)) {
-        // Count quantiles
-        for (const auto &[label, quantiles] : std::get<LabelQuantileVec>(result)) {
-            Json::Value label_obj = get_label_as_json(label);
-            Json::Value quantile_array = Json::arrayValue;
-
-            for (size_t quantile : quantiles) {
-                quantile_array.append(Json::Value((Json::Int64) quantile));
-            }
+    } else if (const auto *v = std::get_if<LabelAbundanceVec>(&result_)) {
+        // k-mer counts (or quantiles)
+        for (const auto &[label, counts] : *v) {
+            assert(counts.size());
+            Json::Value &label_obj = root["results"].append(get_label_as_json(label));
+            Json::Value &counts_array = (label_obj[KMER_ABUNDANCE_FIELD] = Json::arrayValue);
+            if (verbose_output) {
+                for (size_t c : counts) {
+                    counts_array.append(static_cast<Json::Int64>(c));
+                }
+            } else {
+                std::pair<size_t, size_t> last(0, counts.at(0));
+                for (size_t i = 1; i <= counts.size(); ++i) {
+                    if (i < counts.size() && counts[i] == last.second)
+                        continue; // extend run
+
+                    // end of run
+                    if (last.second) {
+                        if (i == last.first + 1) {
+                            counts_array.append(fmt::format("{}={}", last.first, last.second));
+                        } else {
+                            counts_array.append(fmt::format("{}-{}={}", last.first, i - 1, last.second));
+                        }
+                    }
 
-            label_obj[KMER_COUNT_QUANTILE_FIELD] = quantile_array;
-            root["results"].append(label_obj);
+                    // start new run
+                    if (i < counts.size()) {
+                        last.first = i;
+                        last.second = counts[i];
+                    }
+                }
+            }
         }
-    } else if (std::holds_alternative<LabelCoordVec>(result)) {
+    } else {
         // Kmer coordinates
-        for (const auto &[label, tuples] : std::get<LabelCoordVec>(result)) {
-            Json::Value label_obj = get_label_as_json(label);
-            Json::Value coord_array = Json::arrayValue;
+        for (const auto &[label, tuples] : std::get<LabelCoordVec>(result_)) {
```

Comment:
there should at least be an assertion here that `std::get<LabelCoordVec>(result_)` returns a non-`nullptr`, otherwise this'll segfault if we add a new type of annotation later

Reply:
`std::get<LabelCoordVec>` returns a reference or throws an exception.